lighthouse-core/gather/gatherers/styles.js

```diff
@@ -129,13 +138,17 @@ class Styles extends Gatherer {
   afterPass(options) {
     return this.endStylesCollect(options.driver)
       .then(stylesheets => {
-        // Want unique stylesheets. Remove those with the same text content.
+        // Generally want unique stylesheets. Mark those with the same text content.
         // An example where stylesheets are the same is if the user includes a
         // stylesheet more than once (these have unique stylesheet ids according to
         // the DevTools protocol). Another example is many instances of a shadow
         // root that share the same <style> tag.
         const map = new Map(stylesheets.map(s => [s.content, s]));
-        return Array.from(map.values());
+        return stylesheets.map(stylesheet => {
+          const idInMap = map.get(stylesheet.content).header.styleSheetId;
+          stylesheet.isDuplicate = idInMap !== stylesheet.header.styleSheetId;
```

Comment:
I really think we should keep it. https://github.com/GoogleChrome/lighthouse/pull/1421/files#r95644056